src/components/AllocationAddForm.js

```diff
@@ -341,7 +341,7 @@ const AllocationAddForm = (props) => {
             ? dataProjectContributors.allocations
             : null
     const payments = dataClientPayments
-        ? [...dataClientPayments.getProjectById.client.payments, { amount: null, date_paid: null }]
+        ? [...dataClientPayments.getProjectById.client.payments, { amount: null, date_paid: null }].sort((a, b) => b.date_incurred - a.date_incurred)
```

Comment:
This is considered spaghetti code, try to avoid it whenever you can by breaking it into multiple lines like this
```
? [...dataClientPayments.getProjectById.client.payments, { 
     amount: null, 
     date_paid: null 
}].sort((a, b) => b.date_incurred - a.date_incurred)
```
so it can be more readable